src/solver.rs

```diff
@@ -630,7 +664,276 @@ mod tests {
+        );
+        test_invertability(
+            ADD,
+            "*0",
+            0b10,
+            0b11,
+            side,
+            false,
+            "operand with leading undetermined bits",
+        );
+        test_invertability(ADD, "***", 0b100, 0b111, side, true, "undetermined operand");
+        test_invertability(ADD, "***", 0b100, 0b11, side, false, "undetermined operand");
+        test_invertability(
+            ADD,
+            "**0",
+            0b10,
+            0b100,
+            side,
+            true,
+            "operand with multiple undetermined bits",
+        );
+        test_invertability(
+            ADD,
+            "**0",
+            0b1,
+            0b11,
+            side,
+            true,
+            "operand with multiple undetermined bits",
+        );
+        test_invertability(
+            ADD,
+            "**1",
+            0b1,
+            0b111,
+            side,
+            false,
+            "operand with multiple undetermined bits",
+        );
+    }
+
+    #[test]
+    fn compute_inverse_values_for_add() {
+        let side = ArgumentSide::Lhs;
+
+        fn f(l: BitVector, r: BitVector) -> BitVector {
+            l + r
+        }
+
+        // test only for values which are actually invertable
+        test_inverse_value_computation(ADD, "1", 0b1, 0b10, side, f);
+        test_inverse_value_computation(ADD, "0*", 0b10, 0b10, side, f);
+        test_inverse_value_computation(ADD, "***", 0b100, 0b100, side, f);
+        test_inverse_value_computation(ADD, "**0", 0b10, 0b1100, side, f);
+    }
+
+    #[test]
+    fn check_consistency_condition_for_add() {
+        let condition = "t != 0 => x^hi != 0";
+
+        test_consistence(ADD, "1", 0b10, 0b110, true, condition);
+        test_consistence(ADD, "0", 0b10, 0b110, true, condition);
+
+        let condition = "odd(t) => x^hi[lsb] != 0";
+
+        test_consistence(ADD, "*00", 0b1, 0b101, true, condition);
+        test_consistence(ADD, "*01", 0b1, 0b101, true, condition);
+        test_consistence(ADD, "*0*", 0b1, 0b11, true, condition);
+
+        let condition = "Ey.(mcb(x, y) && ctz(t) >= ctz(y))";
+
+        test_consistence(ADD, "*00", 0b1, 0b100, true, condition);
+        test_consistence(ADD, "*00", 0b1, 0b10, true, condition);
+    }
+
+    #[test]
+    fn compute_consistent_values_for_add() {
+        let side = ArgumentSide::Lhs;
+
+        fn f(l: BitVector, r: BitVector) -> BitVector {
+            l + r
+        }
+
+        test_consistent_value_computation(ADD, "0", 0b0, 0b111, side, f);
+        test_consistent_value_computation(ADD, "*01", 0b1, 0b101, side, f);
+        test_consistent_value_computation(ADD, "*0*", 0b1, 0b11, side, f);
+        test_consistent_value_computation(ADD, "*00", 0b1, 0b100, side, f);
+    }
+
+    const SUB: Instruction = Instruction::Sub(RType(0));
+
+    #[test]
+    fn check_invertability_condition_for_sub() {
+        let side = ArgumentSide::Lhs;
```

Comment:
You should definitely test both argument sides, especially because sub is not commutative. I think there is a possible bug in this code, because argument sides are not respected in `compute_consistent_value()`, `compute_inverse_value()` etc.